modules/luci-base/htdocs/luci-static/resources/network.js

```diff
@@ -4436,4 +4460,78 @@ WifiNetwork = baseclass.extend(/** @lends LuCI.network.WifiNetwork.prototype */
 	}
 });
 
+/**
+ * @class
+ * @memberof LuCI.network
+ * @hideconstructor
+ * @classdesc
+ *
+ * A `Network.WifiVlan` class instance represents a vlan on a WifiNetwork.
+ */
+WifiVlan = baseclass.extend(/** @lends LuCI.network.WifiVlan.prototype */ {
+	__init__: function(vlan) {
+		this.ifname = vlan.ifname;
+		if (L.isObject(vlan.config)) {
+			this.vid = vlan.config.vid;
+			this.name = vlan.config.name;
+
+			if (Array.isArray(vlan.config.network))
+				if (vlan.config.network.length)
+					this.network = vlan.config.network[0];
```

Comment:
Feel free to collapse this to one if-statement so we avoid unnecessary nesting.

Reply:
Makes absolutely sense